crates/arrow_sql_gen/src/statement.rs

```diff
@@ -167,6 +167,30 @@ impl InsertBuilder {
                             );
                         }
                     }
+                    DataType::Date32 => {
+                        let array = column.as_any().downcast_ref::<array::Date32Array>();
+                        if let Some(valid_array) = array {
+                            row_values.push(
+                                match OffsetDateTime::from_unix_timestamp(valid_array.value(row) as i64 * 86_400) {
+                                    Ok(offset_time) => offset_time.date().into(),
+                                    Err(_) => return,
```

Comment:
It is weird that we are just ignoring this error - I know this was done elsewhere, but I think we should properly return an error and bubble it up to the runtime to be reported. Otherwise we'll silently ignore these and it could lead to subtle hard to debug data quality bugs.

Fine to create an issue and tackle it in a separate PR though.

Reply:
I've added a Result to `.build()`, and log an error for now.

Reply:
Issue: https://github.com/spiceai/spiceai/issues/1221